macroslib/src/java_jni/jni-include.rs

```diff
@@ -1361,3 +1361,46 @@ impl<T: SwigForeignClass> SwigFrom<Option<T>> for jobject {
         }
     }
 }
+
+#[swig_from_foreigner_hint = "java.util.Optional<T>"]
+impl<T:SwigForeignClass + Clone> SwigFrom<jobject> for Option<T> {
+    fn swig_from(x: jobject, env: *mut JNIEnv) -> Self {
+        let x = unsafe { (**env).NewLocalRef.unwrap()(env, x) };
+        if x.is_null() {
+            None
+        } else {
+            let x = unsafe { (**env).NewLocalRef.unwrap()(env, x) };
+            if x.is_null() {
+                None
+            } else {
+
+                let class_id = <T>::jni_class_name();
```

Comment:
In fact all things like this is expensive (from CPU usage point of view): FindClass, CallMethod etc.

Better would be partly handle it in https://github.com/Dushistov/rust_swig/blob/master/macroslib/src/java_jni/map_type.rs with help of Java not JNI.

Reply:
I don't see anywhere else where that happens. Are there other examples of what you're talking about? I made it work like the other example of `Vec<T>` for now. While inefficient, it seems to work for null and non null cases.